var/spack/repos/builtin/packages/gdbm/package.py

```diff
@@ -30,6 +30,7 @@ class Gdbm(AutotoolsPackage, GNUMirrorPackage):
     patch('gdbm.patch', when='%clang@11:')
     patch('gdbm.patch', when='%cce@11:')
     patch('gdbm.patch', when='%aocc@2:')
+    patch('gdbm.patch', when='%oneapi')
```

Comment:
Not necessary but, given the same patch is applied in at least lines 30-33, this could be changed to a for-loop over the values provided to `when`.